addons/purchase_stock/tests/test_purchase_order.py

```diff
@@ -688,6 +688,29 @@ def test_receive_returned_product_without_po_update(self):
         self.assertEqual(po.order_line[0].qty_received, 5)
         self.assertEqual(po.order_line[1].qty_received, 5)
 
+    def test_receive_negative_quantity(self):
+        """
+        Receive a negative quantity, the picking should be a delivery and the quantity received
+        negative. """
+        po_vals = {
+            'partner_id': self.partner_a.id,
+            'order_line': [Command.create({
+                'name': self.product_id_2.name,
+                'product_id': self.product_id_2.id,
+                'product_qty': 5.0,
+                'product_uom': self.product_id_2.uom_po_id.id,
+                'price_unit': 250.0,
+                'date_planned': datetime.today().replace(hour=9).strftime(DEFAULT_SERVER_DATETIME_FORMAT),
```

Comment:
Out of curiosity: does the date matter in this case?

Reply:
nothing, I removed it